### PR TITLE
Show a proper error message when @Inject is used in NativeTest

### LINKED ISSUE #7300: NativeImageTest that uses @Inject: better error message than NullPointerException
Given a test that runs

```
@QuarkusTest
public class TimeTableResourceTest {

    @Inject
    private TimeTableResource timeTableResource;

    @Test
    public void solveDemoDataUntilFeasible() {
        timeTableResource.solve();
        ...
   }
}
```

The native test version of it:
```

@NativeImageTest
public class TimeTableResourceIT extends TimeTableResourceTest {...}
```

fails because `@Inject` is not supported in native tests:

```
[INFO] Running org.acme.optaplanner.rest.TimeTableResourceIT
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.014 s <<< FAILURE! - in org.acme.optaplanner.rest.TimeTableResourceIT
[ERROR] solveDemoDataUntilFeasible  Time elapsed: 0.013 s  <<< ERROR!
java.lang.NullPointerException
```

**The error message `java.lang.NullPointerException` is not helpful to determine the cause. Can we change that to something like `Don't use @Inject in a native test.`?**

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #7300

### CODE REVIEW

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeTestExtension.java [hunk 1]
```diff
@@ -71,6 +75,23 @@ public void beforeAll(ExtensionContext extensionContext) throws Exception {
...
     }
 
+    private void ensureNoInjectAnnotationIsUsed(Class<?> testClass) {
+        Class<?> current = testClass;
+        while (current.getSuperclass() != null) {
+            for (Field field : current.getDeclaredFields()) {
+                Inject injectAnnotation = field.getAnnotation(Inject.class);
+                if (injectAnnotation != null) {
+                    throw new JUnitException(
+                            "@Inject is not supported in QuarkusTest tests. Offending field is "
```
Comment (Ladicek) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeTestExtension.java]: `in QuarkusTest tests` -> `in NativeImageTest tests`?

Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeTestExtension.java]: Thanks!

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeImageTest.java [hunk 2]
```diff
@@ -20,6 +20,9 @@
  * run, it is expected that the JVM tests will be standard unit tests that are
  * executed by surefire, while the native image tests will be integration tests
  * executed by failsafe.
+ * This also means that injection of beans into a test class using {@code @Inject} is not supported
+ * in native image tests. Such injection is only possible in tests injected with
+ * {@link @QuarkusTest} so the test class structure must be take this into account
```
Comment (gsmet) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeImageTest.java]:
```suggestion
 * {@link @QuarkusTest} so the test class structure must take this into account.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
Looks good except for a small typo.

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeImageTest.java (hunk 2 above)
Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/NativeImageTest.java]: It wouldn't have been a proper PR of mine unless it had a typo :). Thanks for spotting it

**Review by gsmet — APPROVED** [review]

**Pull request merged**